src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -401,9 +401,23 @@ public int MaxResponseHeadersLength
             _curlHandler.Properties :
             _socketsHttpHandler.Properties;
 
-        protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) =>
-            DiagnosticsHandler.IsEnabled() ? _diagnosticsHandler.SendAsync(request, cancellationToken) :
-            _curlHandler != null ? _curlHandler.SendAsync(request, cancellationToken) :
-            _socketsHttpHandler.SendAsync(request, cancellationToken);
+        protected internal override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request), SR.net_http_handler_norequest);
```

Comment:
`request` is already checked for `null` in `HttpClient` prior to calling this; a `Debug.Assert` might be more appropriate here.